### PR TITLE
Added skip hard goal check flag in the rebalance resource

### PR DESCRIPTION
**Pull request by ppatierno**
### Type of change

- Enhancement / new feature

### Description

This PR adds the `skipHardGoalCheck` on the rebalance resource so that the user can provide a resource without any hard goals in the `goals` section and CC will skip the check (default is false).

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/modules/appendix_crds.adoc [hunk 1]
```diff
@@ -2812,10 +2812,12 @@ Used in: xref:type-KafkaClusterRebalance-{context}[`KafkaClusterRebalance`]
 
 [options="header"]
 |====
-|Property        |Description
-|goals    1.2+<.<|A list of goals, ordered by decreasing priority, to use in generating the proposal and executing it.The supported goals are available at https://github.com/linkedin/cruise-control#goals.Providing empty goals means to use all the ones declared in the default.goals cruise control configuration parameter.
+|Property                  |Description
+|goals              1.2+<.<|A list of goals, ordered by decreasing priority, to use in generating the proposal and executing it.The supported goals are available at https://github.com/linkedin/cruise-control#goals.Providing empty goals means to use all the ones declared in the default.goals cruise control configuration parameter.
```
Comment (d-laing) [inline documentation/modules/appendix_crds.adoc]: Please reword the description in the source file. Watch out for the missing spaces!

A list of goals, ordered by decreasing priority, to use for generating and executing the rebalance proposal. The supported goals are available at https://github.com/linkedin/cruise-control#goals. If an empty goals list is provided, the goals declared in the `default.goals` Cruise Control configuration parameter are used.

File: documentation/modules/appendix_crds.adoc [hunk 2]
```diff
@@ -2812,10 +2812,12 @@ Used in: xref:type-KafkaClusterRebalance-{context}[`KafkaClusterRebalance`]
...
 |string array
-|verbose  1.2+<.<|Enable the verbose mode for the JSON string describing the optimization result in the related status.
+|skipHardGoalCheck  1.2+<.<|Whether allow hard goals be skipped in proposal generation. Default is false.
```
Comment (d-laing) [inline documentation/modules/appendix_crds.adoc]: Please reword:

Whether to allow hard goals to be skipped in rebalance proposal generation. Default is false.

**Review by d-laing — APPROVED** [review]
Please make a couple of changes to the descriptions.

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaClusterRebalanceSpec.java [hunk 3]
```diff
@@ -23,20 +23,21 @@
         builderPackage = "io.fabric8.kubernetes.api.builder"
 )
 @JsonInclude(JsonInclude.Include.NON_NULL)
-@JsonPropertyOrder({ "goals", "verbose" })
+@JsonPropertyOrder({ "goals", "skipHardGoalCheck", "verbose" })
```
Comment (tomncooper) [inline api/src/main/java/io/strimzi/api/kafka/model/KafkaClusterRebalanceSpec.java:26]: So we don't currently use the verbose output option in the rebalance request. It is feasible a user may want to see the individual partition reassignments for an optimisation proposal. However, the output can be huge and may overwhelm the status field length limit. I am not saying we need to fix this now but we may need to think of a way to get this information to a user in a K8s friendly way. ConfigMap?

**Review by tomncooper — APPROVED** [review]

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaClusterRebalanceSpec.java (hunk 3 above)
Comment (ppatierno) [reply api/src/main/java/io/strimzi/api/kafka/model/KafkaClusterRebalanceSpec.java:26]: Let's remember to open an issue after RC and after doing some tests to see how the status looks like when verbose is enabled.

**Review by kyguy — APPROVED** [review]

**Pull request merged**